src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/ActivitySource.cs

```diff
@@ -13,16 +14,40 @@ public sealed class ActivitySource : IDisposable
         private static readonly SynchronizedList<ActivityListener> s_allListeners = new SynchronizedList<ActivityListener>();
         private SynchronizedList<ActivityListener>? _listeners;
 
+        /// <summary>
+        /// Construct an ActivitySource object with the input name
+        /// </summary>
+        /// <param name="name">The name of the ActivitySource object</param>
+        public ActivitySource(string name) : this(name, version: "", tags: null) {}
+
         /// <summary>
         /// Construct an ActivitySource object with the input name
         /// </summary>
         /// <param name="name">The name of the ActivitySource object</param>
         /// <param name="version">The version of the component publishing the tracing info.</param>
-        public ActivitySource(string name, string? version = "")
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public ActivitySource(string name, string? version = "") : this(name, version, tags: null) {}
+
+        /// <summary>
+        /// Construct an ActivitySource object with the input name
+        /// </summary>
+        /// <param name="name">The name of the ActivitySource object</param>
+        /// <param name="version">The version of the component publishing the tracing info.</param>
+        /// <param name="tags">The optional ActivitySource tags.</param>
+        public ActivitySource(string name, string? version = "", IEnumerable<KeyValuePair<string, object?>>? tags = default)
         {
             Name = name ?? throw new ArgumentNullException(nameof(name));
             Version = version;
 
+            // Sorting the tags to make sure the tags are always in the same order.
+            // Sorting can help in comparing the tags used for any scenario.
+            if (tags is not null)
+            {
+                var tagList = new List<KeyValuePair<string, object?>>(tags);
+                tagList.Sort((left, right) => string.Compare(left.Key, right.Key, StringComparison.Ordinal));
```

Comment:
what is the usecase for the sorting here?
Also, should this do de-duplication, given Activity Tags are deduped when set via SetTag?

Reply:
> what is the usecase for the sorting here?

We apply the same approach for Meter. The reason is to avoid creating duplicate meters with the same parameters and tags in the cached objects used in DI. While ActivitySource is not currently used in DI, implementing this now ensures consistent behavior and eliminates the need for changes if we decide to support ActivitySource/tracing configuration in the future.

> should this do de-duplication, given Activity Tags are deduped when set via SetTag?

Any reason? Should this be the responsibility of the callers?

Reply:
> should this do de-duplication, given Activity Tags are deduped when set via SetTag?
> 
> Any reason? Should this be the responsibility of the callers? We didn't do that with the Meter tags.

Because Activity.SetTag already does de-duplication, setting some precedence!
For metrics, BCL does not do any deduping of tags!

Reply:
I am fine with this if you see it as important. However, I want to understand the scenario better. Who is going to create a new ActivitySource and potentially have duplicate tags? I don't see this as a likely scenario since the creators of the ActivitySource will know exactly which tags they want to use.